lua/roslyn/slnutils.lua

```diff
@@ -1,5 +1,32 @@
 local M = {}
 
+--- Recursively searches for files with a specific extension within a directory range.
+--- The search starts at the 'from' directory and continues until the 'to' directory.
+--- Only files matching the provided extension are returned.
+---
+--- @param from string The starting directory path for the search.
+--- @param to string? The ending directory path to stop the search. (Cannot contain '/' in the end)
+--- @param extension string The file extension to look for (e.g., ".sln").
+---
+--- @return string[] List of file paths that match the specified extension.
+local function find_files_with_extension(from, to, extension)
```

Comment:
I think we actually can remove the for loop for `parents` here and just have the inner loop and search inside a single directory.

This should be the same as this, since in the `get_solution_files` we are actually first getting the root directory of the first found solution file, so there is no need to search from the buffer to the root directory since we have already done this.

An added benefit of this is that this could be a function that could be used for `try_get_csproj_files` as well I think

Reply:
I think I understood what you mean. Can you check my latest commit?

Reply:
Yes, that is exactly what I was thinking!